src/components/Icon/Icon.scss

```diff
@@ -27,192 +27,394 @@
   @include color-icon('teal');
 
   &::before {
-    background-color: color('white');
+    background-color: var(--p-surface-highlight, color('white'));
+  }
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon-highlight));
   }
 }
 
 .colorTealDark {
   @include color-icon('teal', 'dark');
 
   &::before {
-    background-color: color('teal', 'light');
+    background-color: var(--p-surface-highlight, color('teal', 'light'));
+  }
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon-highlight));
   }
 }
 
 .colorTealDarker {
   @include color-icon('teal', 'darker');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon-highlight));
+  }
 }
 
 .colorGreenLighter {
   @include color-icon('green', 'lighter');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon-success));
+  }
 }
 
 .colorGreen {
   @include color-icon('green');
 
   &::before {
-    background-color: color('green', 'lighter');
+    background-color: var(--p-surface-success, color('green', 'lighter'));
+  }
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon-success));
   }
 }
 
 .colorGreenDark {
   @include color-icon('green', 'dark');
 
   &::before {
-    background-color: color('green', 'light');
+    background-color: var(--p-surface-success, color('green', 'light'));
+  }
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon-success));
   }
 }
 
 .colorYellowLighter {
   @include color-icon('yellow', 'lighter');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon-warning));
+  }
 }
 
 .colorYellow {
   @include color-icon('yellow');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon-warning));
+  }
 }
 
 .colorYellowDark {
   @include color-icon('yellow', 'dark');
 
   &::before {
-    background-color: color('yellow', 'light');
+    background-color: var(--p-surface-warning, color('yellow', 'light'));
+  }
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon-warning));
   }
 }
 
 .colorOrange {
   @include color-icon('orange');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon-warning));
+  }
 }
 
 .colorOrangeDark {
   @include color-icon('orange', 'dark');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon-warning));
+  }
 }
 
 .colorRedLighter {
   @include color-icon('red', 'lighter');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon-critical));
+  }
 }
 
 .colorRed {
   @include color-icon('red');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon-critical));
+  }
 }
 
 .colorRedDark {
   @include color-icon('red', 'dark');
 
   &::before {
-    background-color: color('red', 'light');
+    background-color: var(--p-surface-critical, color('red', 'light'));
+  }
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon-critical));
   }
 }
 
 .colorPurple {
   @include color-icon('purple');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
+}
```

Comment:
Purple does not occur in the new design language, so it is aliased to `--p-icon`.

Reply:
I'd also be comfortable with aliasing this to `highlight`

Reply:
Or primary?

Reply:
Actually yeah looking at the screenshot maybe highlight

Reply:
And also worth surfacing for design review